lib/countly.js

```diff
@@ -1797,6 +1797,236 @@
+                }
+
+                return;
+            }
+        });
+    };
+
+    /**
+    * Present the feedback widget in webview
+    * @param {Object} presentableFeedback - Current presentable feedback
+    **/
+    Countly.present_feedback_widget = function (presentableFeedback) {
+        if (!Countly.check_consent("feedback")) {
+            if (callback) {
+                callback(null, new Error("Consent for feedback not provided."));
+            }
+
+            return;
+        }
+
+        if (!presentableFeedback ||
+            (typeof presentableFeedback !== "object") ||
+            Array.isArray(presentableFeedback)
+        ) {
+            log("Please provide atleast one feedback widget object.");
+            return;
+        }
+
+        try {
+            var url = Countly.url;
+
+            if (presentableFeedback.type === "nps") {
+                url += "/feedback/nps";
+            }
+            else if (presentableFeedback.type === "survey") {
+                url += "/feedback/survey";
+            }
+            else {
+                log("Feedback widget only accepts nps and survey types.");
+                return;
+            }
+
+            url += "?widget_id=" + presentableFeedback._id;
+            url += "&app_key=" + Countly.app_key;
+            url += "&device_id=" + Countly.device_id;
+            url += "&sdk_name=" + SDK_NAME;
+            url += "&platform=" + Countly.platform;
+            url += "&app_version=" + Countly.app_version;
+            url += "&sdk_version=" + SDK_VERSION;
+
+            //Only web SDK passes origin and close
+            url += "&origin=" + window.origin;
+            url += "&close=true";
+            //Origin is passed to the popup so that it passes it back in the postMessage event
+
+            var iframe = document.createElement("iframe");
+            iframe.name = "countly-surveys-iframe";
+            iframe.id = "countly-surveys-iframe";
+            iframe.src = url;
+
+            // iframe.onload = function() {
+            //     document.getElementById('countly-surveys-wrapper-' + presentableFeedback._id).style.display = "none";
+            //     document.getElementById('csbg').style.display = "none";
+            // }
+
+            loadCSS(Countly.url + '/surveys/stylesheets/countly-surveys.css');
+
+            var overlay = document.getElementById("csbg");
+            while (overlay) {
+                //Remove any existing overlays
+                overlay.remove();
+                overlay = document.getElementById("csbg");
+            }
+
+            var wrapper = document.getElementsByClassName("countly-surveys-wrapper");
+            for(var i = 0; i < wrapper.length; i++) {
+                //Remove any existing feedback wrappers
+                wrapper[i].remove();
+            }
+
+            document.body.insertAdjacentHTML('beforeend', '<div id="csbg"></div>');
+
+            var wrapper = document.createElement("div");
```

Comment:
Codacy found an issue: ['wrapper' is already defined.](https://app.codacy.com/manual/ar2rsawseen/countly-sdk-web/pullRequest?prid=6418608)